examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.mm

```diff
@@ -41,10 +42,21 @@ - (void)onStatusUpdate:(CHIPPairingStatus)status
 
 - (void)onPairingComplete:(NSError *)error
 {
+    NSError * __block commissionError;
     CHIP_ERROR err = [CHIPError errorToCHIPErrorCode:error];
+    if (err != CHIP_NO_ERROR) {
+        _commandBridge->SetCommandExitStatus(err);
+        return;
+    }
     ChipLogProgress(chipTool, "Pairing Complete: %s", chip::ErrorStr(err));
-
-    _commandBridge->SetCommandExitStatus(err);
+    [_mCommissioner commissionDevice:_deviceID
```

Comment:
In the "ethernet" case, won't we land here _after_ commissioning is done?  Please check what happens there.

Reply:
Ran through the ethernet scenario and onPairingComplete does not get called and is commissioned successfully.